stardist_napari/_dock_widget.py

```diff
@@ -583,21 +611,64 @@ def progress(it, **kwargs):
                     "image",
                 )
             )
+
+            if model._is_multiclass():
+                prob_class = np.moveaxis(cnn_out[-1], -1, 0)
+                layers.append(
+                    (
+                        prob_class,
+                        dict(
+                            name="StarDist class probabilities",
+                            scale=_scale,
+                            translate=_translate,
+                            **lkwargs,
+                        ),
+                        "image",
+                    )
+                )
         else:
             labels, polys = pred
 
         if output_type in (Output.Labels.value, Output.Both.value):
+
+            if model._is_multiclass() and not (
+                isinstance(model, StarDist3D) and "T" in axes
+            ):
+                #     from skimage.measure import regionprops
+                #     prob, dist, prob_class = cnn_out
+                #     prob_class = np.expand_dims(prob,-1)*prob_class
+                #     labels_cls = np.zeros_like(labels)
+                #     for r in regionprops(labels):
+                #         mask = labels[r.slice]==r.label
+                #         labels_cls[r.slice][mask] = 1+np.argmax(np.sum(prob_class[r.slice][mask], 0)[1:])
+
+                layers.append(
+                    (
+                        labels_multiclass,
+                        dict(
+                            name="StarDist class labels",
+                            visible=False,
+                            scale=scale_out,
+                            opacity=0.5,
+                            **lkwargs,
+                        ),
+                        "labels",
+                    )
+                )
+
             layers.append(
                 (
                     labels,
                     dict(name="StarDist labels", scale=scale_out, opacity=0.5, **lkwargs),
                     "labels",
                 )
             )
+
         if output_type in (Output.Polys.value, Output.Both.value):
```

Comment:
In principle, class labels could be treated like instance labels, i.e. can be shown as label images or polygons / polyhedra.